### PR TITLE
linuxptp: update to v4.4

### PR DESCRIPTION
**Pull request by shenghaoyang**
- Replaced SF project and download URLs with nwtime ones.
  The project is now hosted at https://linuxptp.nwtime.org/.

- Removed 020-gcc14.patch. Missing include was fixed upstream
  in v4.2.

- Added patch to disable MAC library autodetection. That
  can silently pick up unwanted dependencies, depending on
  package build order. We can add linuxptp-<mac lib> variants
  of this package later if there are users of authenticated
  PTP.

Maintainer: @wodu
Compile tested: RB5009 (mvebu/cortexa72 on SNAPSHOT)
Run tested: RB5009 (mvebu/cortexa72 on SNAPSHOT)

Works for implementing a boundary clock using the 88E6393X's PHC.

### CODE REVIEW

**Comment by zpc0** [comment]
> URL:=http://linuxptp.sourceforge.net

Could you also update `URL:=`? Thanks!

**Comment by shenghaoyang** [comment]
> Could you also update URL:=? Thanks!

Thanks for the spot - pushed a fix!

**Comment by BKPepe** [comment]
I checked individual commits and it does not look right.

https://github.com/openwrt/packages/pull/24975/commits/d08dd82740526b95ea3bc1e432e699c4117462a0
-  Here you changed to download tarball not from sourceforge but from their website, but it is said nowhere.

File: net/linuxptp/patches/020-sad-allow-disabling-MAC-library-autodetection.patch [hunk 1]
```diff
@@ -0,0 +1,193 @@
+From 41ed39e7abbdcc68d9ff4baaf022aa2e6d9dadde Mon Sep 17 00:00:00 2001
+From: Shenghao Yang <me@shenghaoyang.info>
+Date: Sun, 15 Sep 2024 14:53:21 +0800
+Subject: [PATCH] sad: allow disabling MAC library autodetection
+
+Signed-off-by: Shenghao Yang <me@shenghaoyang.info>
```
Comment (BKPepe) [inline net/linuxptp/patches/020-sad-allow-disabling-MAC-library-autodetection.patch:6]: Shouldnt be here included commit description from the patch, which you sent to upstream?
https://lists.nwtime.org/sympa/arc/linuxptp-devel/2024-09/msg00051.html

Comment (shenghaoyang) [reply net/linuxptp/patches/020-sad-allow-disabling-MAC-library-autodetection.patch:6]: Ah indeed, used an initial one that didn't have a commit message - let me refresh that patch.

**Comment by shenghaoyang** [comment]
Thanks! I'll update the commit messages and refresh that patch.

**Comment by 1715173329** [comment]
please squash your commits and refresh new added patch
fyi: https://openwrt.org/docs/guide-developer/toolchain/use-patches-with-buildsystem#refreshing_patches

**Comment by shenghaoyang** [comment]
> please squash your commits and refresh new added patch
fyi: https://openwrt.org/docs/guide-developer/toolchain/use-patches-with-buildsystem#refreshing_patches

Thanks! Should have read a bit more - patches were processed with quilt and squashed.

**Comment by shenghaoyang** [comment]
Hmm.. it still thinks they're dirty. Looking through the patches again

**Comment by shenghaoyang** [comment]
:/ It refeshed an older patch and that wasn't included in this commit. Sorting that out...

**Review by systemcrash — APPROVED** [review]

**Pull request merged**